api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/EriDSDCohortQueries.java

```diff
@@ -0,0 +1,1077 @@
+            "startDate=${endDate-12m},endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "CII",
+        EptsReportUtils.map(
+            getCD4CountAndCD4Percent2(),
+            "startDate=${endDate-12m},endDate=${endDate},location=${location}"));
+
+    cd.setCompositionString("(CI OR CII)");
+
+    return cd;
+  }
+
+  /**
+   * 5C (i) One CD4 Lab result > 750 cels/mm3 or > 15% in last ART year (if patients age >=2 and
+   * <=4)
+   *
+   * @return
+   */
+  private CohortDefinition getCD4CountAndCD4Percent1() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+
+    cd.addSearch(
+        "Cd4Abs",
+        EptsReportUtils.map(
+            getValueNumeric(hivMetadata.getCD4AbsoluteOBSConcept(), 750.0),
+            "onOrAfter=${endDate-12m},onOrBefore=${endDate},locationList=${location}"));
+    cd.addSearch(
+        "Cd4Lab",
+        EptsReportUtils.map(
+            getValueNumeric(hivMetadata.getCD4AbsoluteConcept(), 750.0),
+            "onOrAfter=${endDate-12m},onOrBefore=${endDate},locationList=${location}"));
+    cd.addSearch(
+        "Cd4Percent",
+        EptsReportUtils.map(
+            getValueNumeric(hivMetadata.getCD4PercentConcept(), 15.0),
+            "onOrAfter=${endDate-12m},onOrBefore=${endDate},locationList=${location}"));
+    cd.addSearch(
+        "Age",
+        EptsReportUtils.map(
+            ageCohortQueries.createXtoYAgeCohort("2-4", 2, 4), "effectiveDate=${endDate}"));
+
+    cd.setCompositionString("(Cd4Abs OR Cd4Lab) OR (Cd4Percent AND Age)");
+
+    return cd;
+  }
+
+  /**
+   * 5C (ii) One CD4 result > 200 cels/mm3 in last ART year (if patients age >=5 and <=9)
+   *
+   * @return
+   */
+  private CohortDefinition getCD4CountAndCD4Percent2() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+
+    cd.addSearch(
+        "CD4Abs",
+        EptsReportUtils.map(
+            getValueNumeric(hivMetadata.getCD4AbsoluteOBSConcept(), 200.0),
+            "onOrAfter=${endDate-12m},onOrBefore=${endDate},locationList=${location}"));
+    cd.addSearch(
+        "Age",
+        EptsReportUtils.map(
+            ageCohortQueries.createXtoYAgeCohort("5-9", 5, 9), "effectiveDate=${endDate}"));
+
+    cd.setCompositionString("(CD4Abs AND Age)");
+
+    return cd;
+  }
+
+  /**
+   * 5C Generic method to find patients with CD4 Count greater than a given value
+   *
+   * @param concept
+   * @param value
+   * @return
+   */
+  private CohortDefinition getValueNumeric(Concept concept, Double value) {
```

Comment:
Generic cohorts should go into the GenericCohortQueries class to be resuable by other indicators

Reply:
I have changed this as well. @nthfloor